src/engines/kotor/module.cpp

```diff
@@ -701,6 +705,42 @@ void Module::movedPC() {
 		_area->notifyPCMoved();
 }
 
+size_t Module::getPartyMemberCount() {
+	return _party.size();
+}
+
+void Module::addToParty(Creature *creature) {
+	_party.push_back(creature);
+
+	_ingame->setPartyLeader(*_party.begin());
+
+	if (_party.size() == 2) {
+		std::list<Creature *>::iterator iter = _party.begin();
+		std::advance(iter, 1);
+		_ingame->setPartyMember1(*iter);
+	} else {
+		_ingame->setPartyMember1(0);
```

Comment:
This method is completely wrong and broken, and doesn't do what you want it to do.

- So when adding party member 3, party.size() == 3, so you call setPartyMember1(0)?
- Why are you re-setting the party leader each time?
- Why are you iterating over the list when you already have creature?